src/lib/ext-activitylog.js

```diff
@@ -7,11 +7,23 @@ class Model {
       id: [],
       viewType: [],
       type: [],
+      keyword: '',
     };
   }
 
   addNewLogs(logs) {
-    this.logs.push(...logs);
+    for (const log of logs) {
+      this.logs.push(log);
+      this.updateFilter(log);
+    }
+  }
+
+  updateFilter(log) {
```

Comment:
This logic can't be right. This would re-add a disabled filter whenever a new log entry is added.

Reply:
updated :-)

Reply:
@Rob--W Why are we doing the same thing (which brings the same result) in an alternate way that involves more code? 
Also, in terms of performance in the current approach, I imagine that we are monitoring a lot of extensions and for that, we are storing all extensions id, type, viewtype... in the model, and comparing filter matching with `isFilterMatched` method involves more time than it would in the previous solution (where `.hidden = filter(log)`).

Reply:
> @Rob--W Why are we doing the same thing (which brings the same result) in an alternate way that involves more code?

What part of the alternative is required and more code according to you?

In my view, the `addFilter` and `removeFilter` functions are not necessary. When they were added, it was to support the identity filtering method, but it would never work for the substring filtering method. A "filterchange" event that shares the current value of the filter (i.e. the full filter opposed to the addition or removal) would work for the checkbox and keyword input. There is no need for several more events.

I'm kind of surprised to see the introduction of the lot of events here, and am glad that you are questioning the need for them.

> Also, in terms of performance in the current approach, I imagine that we are monitoring a lot of extensions and for that, we are storing all extensions id, type, viewtype... in the model, and comparing filter matching with `isFilterMatched` method involves more time than it would in the previous solution (where `.hidden = filter(log)`).

I don't understand this claim. Why do you believe that there is a significant performance difference between the two?

Reply:
> > @Rob--W Why are we doing the same thing (which brings the same result) in an alternate way that involves more code?
> 
> What part of the alternative is required and more code according to you?

In previous logic, I didn't need to setup a communication (which I am doing currently [here](https://github.com/mozilla/extension-activity-monitor/pull/14/files#diff-9f25b3c1ffafaec3c4d02ac711c6392aR33-R37) between `filter-option-element` and model to update the model when a new checkbox is added. 

> In my view, the `addFilter` and `removeFilter` functions are not necessary. When they were added, it was to support the identity filtering method, but it would never work for the substring filtering method. A "filterchange" event that shares the current value of the filter (i.e. the full filter opposed to the addition or removal) would work for the checkbox and keyword input. There is no need for several more events.

If I remove `addFilter` and `removeFilter` functions, how do I modify the model's filter object? Modification to filter object in model is required since it says if any of the future logs should hidden or not. 

> > Also, in terms of performance in the current approach, I imagine that we are monitoring a lot of extensions and for that, we are storing all extensions id, type, viewtype... in the model, and comparing filter matching with `isFilterMatched` method involves more time than it would in the previous solution (where `.hidden = filter(log)`).
> 
> I don't understand this claim. Why do you believe that there is a significant performance difference between the two?

There might not be significant performance difference is practical but I was thinking, in the current approach where we save all the ids, viewTypes, type, api names and searching inside them. But previously logic, the filter object would contain only ids, viewTypes, type, api names of those which we want to be hidden. So, there would be less items (unless anyone filters out everything) in the filter object compared to current approach, and searching inside them would be faster for that?

Reply:
> > > @Rob--W Why are we doing the same thing (which brings the same result) in an alternate way that involves more code?
> > 
> > 
> > What part of the alternative is required and more code according to you?
> 
> In previous logic, I didn't need to setup a communication (which I am doing currently [here](https://github.com/mozilla/extension-activity-monitor/pull/14/files#diff-9f25b3c1ffafaec3c4d02ac711c6392aR33-R37) between `filter-option-element` and model to update the model when a new checkbox is added.

This brings up an interesting point: on what is expected to happen when a new log entry is added. When the user isn't filtering anything, the expected result is that the option should be checked by default (this is what you have in mind). But there are use cases where the user is looking for specific API calls, and in that case it would be preferred that new logs that don't match the existing criteria are hidden by default. The two different use cases can be supported by adding a checkbox `[ ] Select all by default` to change whether new options should be checked (or unchecked) by default. This can be done in a follow-up (consider filing an issue in this repo if you want to work on it later).

> > In my view, the `addFilter` and `removeFilter` functions are not necessary. When they were added, it was to support the identity filtering method, but it would never work for the substring filtering method. A "filterchange" event that shares the current value of the filter (i.e. the full filter opposed to the addition or removal) would work for the checkbox and keyword input. There is no need for several more events.
> 
> If I remove `addFilter` and `removeFilter` functions, how do I modify the model's filter object? Modification to filter object in model is required since it says if any of the future logs should be hidden or not.

The filter-option-element has a complete picture of all relevant checkboxes. It could simply send all relevant options to the model (so the model won't have to do `indexOf` + `splice` for example). The controller then doesn't need to know about the format of the filter any more, every `filterchange` event would then replace the filter.

> > > Also, in terms of performance in the current approach, I imagine that we are monitoring a lot of extensions and for that, we are storing all extensions id, type, viewtype... in the model, and comparing filter matching with `isFilterMatched` method involves more time than it would in the previous solution (where `.hidden = filter(log)`).
> > 
> > 
> > I don't understand this claim. Why do you believe that there is a significant performance difference between the two?
> 
> There might not be significant performance difference is practical but I was thinking, in the current approach where we save all the ids, viewTypes, type, api names and searching inside them. But previously logic, the filter object would contain only ids, viewTypes, type, api names of those which we want to be hidden. So, there would be less items (unless anyone filters out everything) in the filter object compared to current approach, and searching inside them would be faster for that?

This is indeed valid for the use case of "show all logs except for X", but doesn't work for "show logs matching Y".
At this stage, it's important to get the prototype working first, optimizations can be done in a follow-up, once it's clear what you need to optimize.

Note: use `Set` instead of `Array` for faster lookups.

Reply:
done :-)

Reply:
done :-)